blockchain/blockchain.go

```diff
@@ -975,6 +985,13 @@ func (b *Blockchain) WriteBlock(block *types.Block, source string) error {
 		logArgs = append(logArgs, "generation_time_in_seconds", diff)
 	}
 
+	b.writeLock.Lock()
```

Comment:
Why is this lock moved here?

Reply:
That should be at the beginning of the function. I'll move it.